core/controllers/dashboard_test.py

```diff
@@ -389,30 +416,204 @@ def test_stats_for_multiple_explorations_with_multiple_owners(self):
+
+        rights_manager.assign_role_for_exploration(
+            self.owner_id_1, self.EXP_ID_1, self.owner_id_2,
+            rights_manager.ROLE_OWNER)
+
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(len(response['explorations_list']), 1)
+
+        exp_version = self.EXP_DEFAULT_VERSION
+        exp_id = self.EXP_ID_1
+        state = exploration.init_state_name
+
+        self._record_start(exp_id, exp_version, state)
+        self._record_start(exp_id, exp_version, state)
+        self._record_exploration_rating(exp_id, [3, 4, 5])
+
+        self.logout()
+
+        self.login(self.OWNER_EMAIL_2)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(len(response['explorations_list']), 1)
+
+        self._record_exploration_rating(exp_id, [1, 5, 4])
+
+        expected_results = {
+            'total_plays': 2,
+            'num_ratings': 6,
+            'average_ratings': 22/6.0
+        }
+
+        user_stats_1 = (
+            user_jobs_continuous.UserStatsAggregator.get_dashboard_stats(
+                self.owner_id_1))
+        self.assertEquals(
+            user_stats_1['total_plays'], expected_results['total_plays'])
+        self.assertEquals(
+            user_stats_1['num_ratings'], expected_results['num_ratings'])
+        self.assertEquals(
+            user_stats_1['average_ratings'],
+            expected_results['average_ratings'])
+
+        user_stats_2 = (
+            user_jobs_continuous.UserStatsAggregator.get_dashboard_stats(
+                self.owner_id_2))
+        self.assertEquals(
+            user_stats_2['total_plays'], expected_results['total_plays'])
+        self.assertEquals(
+            user_stats_2['num_ratings'], expected_results['num_ratings'])
+        self.assertEquals(
+            user_stats_2['average_ratings'],
+            expected_results['average_ratings'])
+
+        self.logout()
+
+    def test_realtime_layer_batch_job_multiple_explorations_one_owner(self):
+        self.save_new_default_exploration(
+            self.EXP_ID_1, self.owner_id_1, title=self.EXP_TITLE_1)
+        self.save_new_default_exploration(
+            self.EXP_ID_2, self.owner_id_1, title=self.EXP_TITLE_2)
+
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(len(response['explorations_list']), 2)
+
+        exp_id_1 = self.EXP_ID_1
+        exp_id_2 = self.EXP_ID_2
+
+        self._record_exploration_rating(exp_id_1, [4, 5, 2])
+        self._record_exploration_rating(exp_id_2, [5, 2])
+
+        user_stats = (
+            user_jobs_continuous.UserStatsAggregator.get_dashboard_stats(
+                self.owner_id_1))
+        self.assertEquals(user_stats['total_plays'], 0)
+        self.assertEquals(user_stats['num_ratings'], 5)
+        self.assertEquals(user_stats['average_ratings'], 18/5.0)
```

Comment:
Initially, it was the former. But now that the number of ratings are being stored in the user stats model itself, calculating average_ratings through the latter method makes sense since the latter take into consideration the popularity of an exploration as well!

For example,
A creator has 2 published explorations, the 1st one has been rated four times - 3, 4, 4, 5 . And the 2nd one has been rated only once - 2.
Then calculating the average rating by simply averaging over the average ratings of each exploration yields 3. While if the average ratings value is calculated by summation of all ratings by num of ratings  then it comes out to be 3.6, which IMO is a better portrayal of the scenario!